validate/validators/bam.py

```diff
@@ -9,4 +9,8 @@ def validate_bam_file(path):
     except pysam.SamtoolsError as err:
         raise ValueError("samtools bam check failed. " + str(err)) from err
 
+    bam_it = pysam.AlignmentFile(str(path)).head(1)
```

Comment:
In think generally, we want to avoid having the type of a variable in the variable name, right @tyamaguchi-ucla @aholmes?

Reply:
It depends, however I do personally try to avoid it. Here, I would use a type hint. The pysam docs say it is a `IteratorRowHead`, so something like this, perhaps.

```python
bam_head: IteratorRowHead = pysam.AlignmentFile(str(path)).head(1)
```